datadog_lambda/metric.py

```diff
@@ -22,26 +24,42 @@ def _format_dd_lambda_layer_tag():
     return "dd_lambda_layer:datadog-{}_{}".format(runtime, __version__)
 
 
-def _tag_dd_lambda_layer(args, kwargs):
+def _tag_dd_lambda_layer(tags):
     """
     Used by lambda_metric to insert the dd_lambda_layer tag
     """
     dd_lambda_layer_tag = _format_dd_lambda_layer_tag()
-    if 'tags' in kwargs:
-        kwargs['tags'].append(dd_lambda_layer_tag)
-    elif len(args) >= 4:
-        args[3].append(dd_lambda_layer_tag)
+    if tags:
+        return tags + [dd_lambda_layer_tag]
     else:
-        kwargs['tags'] = [dd_lambda_layer_tag]
+        return [dd_lambda_layer_tag]
 
 
-def lambda_metric(*args, **kwargs):
+def lambda_metric(metric_name, value, timestamp=None, tags=None):
     """
     Submit a data point to Datadog distribution metrics.
     https://docs.datadoghq.com/graphing/metrics/distributions/
+
+    When DATADOG_LOG_FORWARDER is True, write metric to log, and
+    wait for the Datadog Log Forwarder Lambda function to submit
+    the metrics asynchronously.
+
+    Otherwise, the metrics will be submitted to the Datadog API
+    periodically and at the end of the function execution in a
+    background thread.
     """
-    _tag_dd_lambda_layer(args, kwargs)
-    lambda_stats.distribution(*args, **kwargs)
+    tags = _tag_dd_lambda_layer(tags)
+    if os.environ.get('DATADOG_FLUSH_TO_LOG') == 'True':
+        print(json.dumps({
+            'metric_name': metric_name,
+            'value': value,
```

Comment:
So, with the golang version of this, it would be simpler if I could send this as 'values:[value]', otherwise I'm going to have to write another serializer/marshaller. It also makes it compatible with the /series api endpoint. Is it a trivial change to do that, or does changing the log crawler function make it complicated?

Reply:
It's not a difficult code change to make, but it results in confusing code (e.g., `values: value`). Also, I'm calling this very `lambda_metric` function in log forwarder with the json writting to log, e.g., I can do simply do `lambda_metric(**metric_json)` with the current consistent signature.

BTW, while reading dd-lambda-go again, how are you preserving the actual `Timestamp` when the metric was submitted? I don't see `Timestamp` field in your struct for instance. Do they get timestamped when it's flushed? I

Reply:
No, the timestamping happens when the "Distribution" function is called. The "series" api takes a list of values as `values:[[$timestamp1, metricValue1], [$timestamp1, metricValue1]]`.

Reply:
My mistake, it uses "points" instead of values. Looks like it's incompatible enough that it doesn't make sense to try and make it match the API.
https://docs.datadoghq.com/api/?lang=bash#post-timeseries-points

Reply:
And, looking at it again, you're right, I think the timestamp isn't being saved. Damn, I'll go fix it.